apps/condo/domains/resident/schema/SendResidentMessageService.js

```diff
@@ -0,0 +1,104 @@
+ * @param data
+ * @returns {Promise<boolean>}
+ */
+const validateResidentMessageData = async (context, data) => {
+    const { organizationId, propertyDetails, type, data: notificationData = {} } = data
+    const { categoryId } = notificationData
+
+    // Validate notification type
+    if (!MESSAGE_META[type]) throw new GQLError(ERRORS.INVALID_NOTIFICATION_TYPE_PROVIDED, context)
+
+    if (isEmpty(propertyDetails)) throw new GQLError(ERRORS.PROPERTY_DETAILS_IS_EMPTY, context)
+
+    // Validate property/billingProperty existance for each details item
+    for (const detail of propertyDetails) {
+        if (!detail.propertyId && !detail.billingPropertyId) throw new GQLError(ERRORS.PROPERTY_IS_REQUIRED, context)
+    }
+
+    // validate organization id
+    const organizationData = await Organization.getOne(context, { id: organizationId, deletedAt: null })
+
+    if (isEmpty(organizationData)) throw new GQLError(ERRORS.INVALID_ORGANIZATION_PROVIDED, context)
+
+    if (!isEmpty(categoryId)) {
+        // validate billing category id
+        const categoryData = await BillingCategory.getAll(context, { id: categoryId, deletedAt: null })
+
+        if (isEmpty(categoryData)) throw new GQLError(ERRORS.INVALID_CATEGORY_PROVIDED, context)
+    }
+}
+
+const SendResidentMessageService = new GQLCustomSchema('SendResidentMessageService', {
+    types: [
+        {
+            access: true,
+            type: 'input PropertyUnitInput { unitType: String, unitName: String }',
+        },
+        {
+            access: true,
+            type: 'input PropertyDetailsInput { propertyId: ID, billingPropertyId: ID, sections: [Int], units: [PropertyUnitInput], phones: [String], billingAccounts: [ID] }',
+        },
+        {
+            access: true,
+            type: 'input SendResidentMessageDataInput { title: String, message: String, urlTemplate: String, categoryId: String, period: String }',
+        },
+        {
+            access: true,
+            type: 'input SendResidentMessageInput { dv: Int!, sender: JSON!, organizationId: ID!, propertyDetails: [PropertyDetailsInput]!, uniqKeyTemplate: String, type: String!, data: SendResidentMessageDataInput }',
+        },
+        {
+            access: true,
+            type: 'type SendResidentMessageOutput { status: String! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.canSendResidentMessage,
+            schema: 'sendResidentMessage(data: SendResidentMessageInput!): SendResidentMessageOutput',
+            doc: {
+                summary: 'Sends notification to all residents of organization properties',
+                errors: ERRORS,
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+
+                await validateResidentMessageData(context, data)
+                await sendResidentMessageTask.delay(JSON.stringify(data))
+
+                return { status: SUCCESS_STATUS }
+            },
```

Comment:
why do really need this? it's easy to just run `sendResidentMessageTask`

Reply:
We need service to be able to trigger it from outside of condo, for example from EPS.